test/DAICO.test.js

```diff
@@ -33,85 +33,188 @@ contract('DAICO', function (accounts) {
     });
 
-    it('owner can t withdraw more than tap times time', async function () {
-      let owner = accounts[0];
+    it('no one can withdraw if she is not owner', async function () {
       let no_owner = accounts[3];
       await this.daico.buyTokens(no_owner, {from: no_owner, value: higher_value}).should.be.fulfilled;
       await increaseTimeTo(this.afterlastWithdrawn);
-      const initial_balance = web3.eth.getBalance(owner).c[0];
-      const timestamp = web3.eth.getBlock(web3.eth.blockNumber).timestamp;
-      await this.daico.withdraw({from: owner});
-      const final_balance = web3.eth.getBalance(owner).c[0];
-      assert.ok(final_balance - initial_balance <= tap * (timestamp - this.lastWithdrawn))
+      await this.daico.withdraw({from: no_owner}).should.be.rejectedWith(EVMRevert);
     });
+
   });
 
-  describe('voting as expected', function () {
+  describe('proposals', function () {
     it('no purchaser no holder', async function() {
       let purchaser = accounts[5];
       let isholder = await this.daico.isHolder(purchaser);
       assert.ok(!isholder);
     });
 
-    it('if yoy did not purchased a token you are not a holder', async function() {
+    it('if you did not purchased a token you are not a holder', async function() {
       let purchaser = accounts[4];
       await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value}).should.be.fulfilled;
       let isholder = await this.daico.isHolder(purchaser);
       assert.ok(isholder);
     });
 
-    it('holder can propose a new tap', async function() {
+    it('holder can propose a new tap in tap mode', async function() {
       let purchaser = accounts[4];
       await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value}).should.be.fulfilled;
+      await increaseTimeTo(this.afterlastWithdrawn);
       await this.daico.newRaiseTapProposal(tap * 2, 3600, {from: purchaser}).should.be.fulfilled;
     });
 
     it('if you are not a holder you cant propose a new tap', async function() {
       let noholder = accounts[4];
+      await increaseTimeTo(this.afterlastWithdrawn);
       await this.daico.newRaiseTapProposal(tap * 2, 3600, {from: noholder}).should.be.rejectedWith(EVMRevert);
     });
 
-    it('you can not propose lower the tap', async function() {
+    it('holder can vote a proposal in time', async function() {
       let purchaser = accounts[4];
-      await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value}).should.be.fulfilled;
-      await this.daico.newRaiseTapProposal(100, 3600, {from: purchaser}).should.be.rejectedWith(EVMRevert);
+      await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value});
+      await increaseTimeTo(this.afterlastWithdrawn);
+      await this.daico.newRaiseTapProposal(100, 3600, {from: purchaser}).should.be.fulfilled;
+      await this.daico.vote(0, true, {from: purchaser}).should.be.fulfilled;
```

Comment:
we need to check whatever you're expecting to happen from the contract side after voting

Reply:
I did this in `00663f0` commit